src/components/Video.astro

```diff
@@ -0,0 +1,32 @@
+---
+import { getLanguageFromURL } from '../util';
+import languages from '../i18n/languages';
+
+export interface Props {
+	src: string;
+	type: string;
+}
+
+const { src, type } = Astro.props as Props;
+const url = new URL(Astro.request.url);
+const lang = getLanguageFromURL(url.pathname);
+const filename = src.substring(src.lastIndexOf('/') + 1, src.lastIndexOf('.'));
```

Comment:
What do you think of using [Node’s path parser](https://nodejs.org/dist/latest-v14.x/docs/api/path.html#path_path_parse_path) here? Should work on URLs too in my testing.

```suggestion
const filename = parse(src).name;
```

Reply:
That's so nice! Didn't know Node had this method, thanks so much! 💜